### DISCUSSION TITLE
HTML, CSS website

### DISCUSSION THREAD

**Discussion by ghost** [community/community#54566] [Programming Help] [answered]
### Body

Hi guys, i have encounter a problem at my html and css code. I found out that if i use the email type input at the html, it cannot run the css on my website :(

the html code : 
```
<div class="loginDetails">
                    <span class="icon">
                        <ion-icon name="mail"></ion-icon>
                    </span>
                    <input type="email" required>
                    <label>Email</label>
                </div>
  ```
     
the css code:
.loginDetails input:focus~label {
    top: -5px;
}

.loginDetails input:valid~label {
    top: -5px;
}

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by airtower-luna** [comment]
Under "the html code" it just says "Email". You need to [use a Markdown code block](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) in your post so it stays readable.

**Reply by ghost** [reply]
thank you for noticing me ! now the code is visible.

**Comment by VikingDDM** [comment]
Need to be provided more code related to it. It just has no problem at this moment. Maybe the other css styles affect to it.

**Reply by ghost** [reply]
should i post the entire source code in here ?

**Reply by VikingDDM** [reply]
Can I see the whole code of this page? I am gonna run code on my side.

**Reply by ghost** [reply]
sure i send the file here?

**Reply by VikingDDM** [reply]
Yes.

**Reply by ghost** [reply]
[loginReg.zip](https://github.com/community/community/files/11412545/loginReg.zip)

**Reply by VikingDDM** [reply]
I can't recognize the issue only with it. Maybe the css style top would be cause something. So I wanna know the parent's style.

**Reply by VikingDDM** [reply]
Ohh! I have received. I'll check it.

**Reply by ghost** [reply]
what is parent's style ?

**Reply by VikingDDM** [reply]
https://user-images.githubusercontent.com/119115648/236633137-7ac89770-0579-4251-bc48-3b7062c8c200.mp4

**Reply by VikingDDM** [reply]
[1.zip](https://github.com/community/community/files/11412676/1.zip)

**Reply by ghost** [reply]
is it a video? i cant play it.

**Reply by ghost** [reply]
> [1.zip](https://github.com/community/community/files/11412676/1.zip)

oh ya i have watched it. Is there a problem at the password field?

**Reply by VikingDDM** [reply]
no.

**Reply by ghost** [reply]
Then the video is for?

**Reply by VikingDDM** [reply]
Just your purpose. what's wrong here? top:-5px works well.

**Reply by ghost** [reply]
> Just your purpose. what's wrong here? top:-5px works well.

it works well at the password field but it doesnt work well at the email field :(

**Reply by VikingDDM** [reply]
Um, can you me your screen capture? The email field also works well on my side.

**Reply by ghost** [reply]
> Um, can you me your screen capture? The email field also works well on my side.
![image](https://user-images.githubusercontent.com/104344742/236634153-cb2aa172-159b-4b23-ac4c-66f13d11c5cc.png)

![image](https://user-images.githubusercontent.com/104344742/236634162-69124374-bf4a-4b23-9659-83f90a581560.png)

**Reply by VikingDDM** [reply]
I got the point.

**Reply by VikingDDM** [reply]
And the solution. I will send you the code and the capture.

**Reply by ghost** [reply]
> And the solution. I will send you the code the capture.

thank you so much !
and i have encounter another problem : D 
can you also help me debug?

**Reply by VikingDDM** [reply]
[2.zip](https://github.com/community/community/files/11412770/2.zip)
![Annotation 2023-05-06 181016](https://user-images.githubusercontent.com/119115648/236635668-bc32a8c3-5237-4c16-9897-699046dd7fb6.png)
![Annotation 2023-05-06 181108](https://user-images.githubusercontent.com/119115648/236635672-bf3a7ed6-f046-409b-a4f0-0bb03ab9a9ed.png)

**Reply by ghost** [reply]
> [2.zip](https://github.com/community/community/files/11412770/2.zip) ![Annotation 2023-05-06 181016](https://user-images.githubusercontent.com/119115648/236635668-bc32a8c3-5237-4c16-9897-699046dd7fb6.png) ![Annotation 2023-05-06 181108](https://user-images.githubusercontent.com/119115648/236635672-bf3a7ed6-f046-409b-a4f0-0bb03ab9a9ed.png)

thank you so much 🙏

**Reply by ghost** [reply]
> And the solution. I will send you the code and the capture.

Another problem is my js cant validate the username and password although i had enter the correct one. :(

**Reply by VikingDDM** [reply]
Urrrr. Question. I got the problem. then send me the exact validate condition. the name and password. So the name field should include both number and letter?

**Reply by ghost** [reply]
> Urrrr. Question. I got the problem. then send me the exact validate condition. the name and password. So the name field should include both number and latter?

Is the same file I sent u just now : D. The name and password can be inspect through the web browser. Im using Google chrome to inspect.

**Reply by ghost** [reply]
> > Urrrr. Question. I got the problem. then send me the exact validate condition. the name and password. So the name field should include both number and latter?
> 
> Is the same file I sent u just now : D. The name and password can be inspect through the web browser. Im using Google chrome to inspect.

![image](https://user-images.githubusercontent.com/104344742/236636668-6bca87b4-7c56-4f87-8338-a8a6307d2742.png)

click the application

**Reply by ghost** [reply]
> > > Urrrr. Question. I got the problem. then send me the exact validate condition. the name and password. So the name field should include both number and latter?
> > 
> > 
> > Is the same file I sent u just now : D. The name and password can be inspect through the web browser. Im using Google chrome to inspect.
> 
> ![image](https://user-images.githubusercontent.com/104344742/236636668-6bca87b4-7c56-4f87-8338-a8a6307d2742.png)
> 
> click the application

![image](https://user-images.githubusercontent.com/104344742/236636700-faf24ec5-ef4c-46f9-976f-62406302ec49.png)

then the username and password will store in the local storage after register

**Reply by VikingDDM** [reply]
I solved it.

**Reply by ghost** [reply]
> I solved it.

Is the mistake is from the loop?

**Reply by VikingDDM** [reply]
I'll send the code.

**Reply by VikingDDM** [reply]
[3.zip](https://github.com/community/community/files/11412933/3.zip)
![Annotation 2023-05-06 185610](https://user-images.githubusercontent.com/119115648/236637413-05ac909d-3883-4885-92f7-07490abab334.png)

**Reply by VikingDDM** [reply]
There was some problem in using conditional statement - JavaScript.

**Reply by ghost** [reply]
> There was some problem in using conditional statement - JavaScript.

Oh i see, but how about the login page?  It can't validate the registered username and password although i have entered the correct pair of the username and password.

**Reply by ghost** [reply]
> > There was some problem in using conditional statement - JavaScript.
> 
> Oh i see, but how about the login page? It can't validate the registered username and password although i have entered the correct pair of the username and password.

![image](https://user-images.githubusercontent.com/104344742/236637668-67b80f64-11b4-4f0d-b2ec-6f05a5f50f73.png)
![image](https://user-images.githubusercontent.com/104344742/236637675-6c0b2066-5320-48d0-9141-38414044f792.png)

**Reply by VikingDDM** [reply]
I got it. so the solution is... The local storage data type is array, isn't it?

**Reply by ghost** [reply]
> > I got it. so the solution is... The local storage data type is array, isn't it?
> 
>Ya is array

**Reply by ghost** [reply]
> I got it. so the solution is... The local storage data type is array, isn't it?

I need to get data from the registration there and store it for login used

**Reply by VikingDDM** [reply]
I have done.

**Reply by VikingDDM** [reply]
So I have fixed a little in existenceCheck()

**Reply by VikingDDM** [reply, marked as answer]
`function existenceCheck() {
    var custName = document.getElementById("iname").value;
    var custPass = document.getElementById("ipassword").value;
    
    userInfoStored = JSON.parse(localStorage.getItem('username'));

    for(let i = 0; i < userInfoStored.length; i++) {
        
        if(custName == userInfoStored[i].ursname && custPass == userInfoStored[i].password) {
            window.location.assign("home.html");
            alert('Login Successfully');
            break;
        } else if(i == userInfoStored.length-1) {
            alert('Invalid username and password.'); 
            break;
        }
    }
}`

**Reply by VikingDDM** [reply]
I'll work.

**Reply by ghost** [reply]
> `function existenceCheck() { var custName = document.getElementById("iname").value; var custPass = document.getElementById("ipassword").value;
> 
> ```
> userInfoStored = JSON.parse(localStorage.getItem('username'));
> 
> for(let i = 0; i < userInfoStored.length; i++) {
>     
>     if(custName == userInfoStored[i].ursname && custPass == userInfoStored[i].password) {
>         window.location.assign("home.html");
>         alert('Login Successfully');
>         break;
>     } else if(i == userInfoStored.length-1) {
>         alert('Invalid username and password.'); 
>         break;
>     }
> }
> ```
> 
> }`

Thank you so much. I know whats my code mistake :D !

**Reply by VikingDDM** [reply]
That's all right.

**Reply by VikingDDM** [reply]
why is it still unanswered?
